### PR TITLE
improve views positioning, rerender on screen resize

### PR DESCRIPTION
**Pull request by alexkuz**
* improve views positioning

Positioning with percentage doesn't work very well for arbitrary width/height. This fix calculates position manually and makes sure there are no gaps between views.

* ~~Optional statusbar view~~

~~This view shows lines from stdout or stderr that matches pattern defined with `--statuspattern` option. For example, it can be used for showing builder progress. Status view is visible only if `--statuspattern` is defined.~~

image

### CODE REVIEW

**Comment by jasonwilson** [comment]
Thanks for the PR @alexkuz. Can you describe other use cases beyond webpack where the statusbar would be valuable? Trying to understand if this could be generalized as filtering.

**Comment by alexkuz** [comment]
I think it could be useful for any prolonged process - project building (not necessarily webpack, of course), file downloading, etc. If there is either too many data that you don't want to litter your log (like download progress), or when you want to trace some process that is hard to follow otherwise, for example, because of a lot of logging (like project building steps).

But I have to say I made it with webpack in mind, so yeah, maybe it's too specific. Not sure how we could generalize it though.

**Comment by jasonwilson** [comment]
@alexkuz would you mind pulling out the view positioning calculations into a separate PR? I'm happy to accept those (nice work btw) and want to spend some time thinking about filtering.

**Comment by alexkuz** [comment]
Sure, I thought these should be separate PRs too, just had no time to do it.

**Comment by alexkuz** [comment]
@jasonwilson I'm leaving just positioning here, also it reattaches internal views on resize now, since `blessed-contrib` doesn't seem to handle resize properly.

**Comment by alexkuz** [comment]
I'll add another PR with status later.

File: lib/dashboard.js [hunk 1]
```diff
@@ -30,41 +30,94 @@ Dashboard.prototype.onEvent = function (event) {
...
+  return {
+    top: top,
+    height: bottom - top,
+    left: Math.ceil(parent.width * stdOutWidth),
+    width: Math.floor(parent.width * (1 - stdOutWidth))
+  };
+};
+
+Dashboard.prototype._logStream = function (streamView, data) {
+  var lines = data.replace(/\n$/, "");
```
Comment (jasonwilson) [inline lib/dashboard.js]: What is the behavior if a chunk contained multiple new line characters?

Comment (aisapatino) [inline lib/dashboard.js]: i think it'd be good to put this in a separate PR since it's not related to the rest of these changes

and maybe have it enabled by an option so the default behavior stays the same - @jasonwilson what do you think?

File: package.json [hunk 2]
```diff
@@ -14,6 +14,7 @@
   "scripts": {
     "lint": "eslint .",
     "test": "mocha -c --recursive test/**/*.spec.js",
+    "test:app": "./bin/nodejs-dashboard.js node test/app/index.js",
```
Comment (aisapatino) [inline package.json]: 👍

File: lib/dashboard.js (hunk 1 above)
Comment (aisapatino) [inline lib/dashboard.js]: this only strips one newline char at the end. i tried it out, things with embedded newlines (like errors) look fine

Comment (jasonwilson) [inline lib/dashboard.js]: 👍

Comment (aisapatino) [inline lib/dashboard.js]: (disregard ^)

Comment (alexkuz) [inline lib/dashboard.js]: Yes, sorry I forgot to make it clear - that last `\n` generated redundant new line, that didn't occur if you run app without dashboard, so I added this fix for consistency.
